Makefile

```diff
@@ -36,6 +36,9 @@ TELEPORT_DEBUG ?= false
 GITTAG=v$(VERSION)
 CGOFLAG ?= CGO_ENABLED=1
 KUSTOMIZE_NO_DYNAMIC_PLUGIN ?= kustomize_disable_go_plugin_support
+KUBECTL_VERSION := $(shell go run ./build.assets/kubectl-version/main.go)
```

Comment:
This is expensive to run each time make is invoked. An alternate command to run that I think would be cheaper is:

    KUBECTL_VERSION := $(shell go mod list -m -f '{{.Version}}' k8s.io/kubectl | sed 's/v0/v1/')

It is still not ideal that this would run every time `make` is invoked, but should be significantly cheaper than building a go program that imports k8s.io modules.